src/index.ts

```diff
@@ -137,14 +144,18 @@ class Egnyte {
      * @param password the passworf for auth
      * @returns the auth cookie string
      */
-    private async _authenticate(): Promise<string> {
-        const csrfToken = await this._getCsrfToken()
-        const auth = await this._egnyteRequest('/accounts/login/?next=/customer/browse/', {
+    private async _authenticate(): Promise<any> {
```

Comment:
Not a big deal at all, but I personally prefer to use destructuring syntax when possible to avoid having to use dot-notation down the line. E.g., `const { csrfMiddlewareToken, csrfToken } = await this._getCsrfTokens()`.